model/client4.go

```diff
@@ -1782,8 +1782,12 @@ func (c *Client4) DeleteChannel(channelId string) (bool, *Response) {
 }
 
 // GetChannelByName returns a channel based on the provided channel name and team id strings.
-func (c *Client4) GetChannelByName(channelName, teamId string, etag string) (*Channel, *Response) {
-	if r, err := c.DoApiGet(c.GetChannelByNameRoute(channelName, teamId), etag); err != nil {
+func (c *Client4) GetChannelByName(channelName, teamId string, etag string, includeDeleted bool) (*Channel, *Response) {
```

Comment:
I think this might be considered a break in backwards compatability?

Reply:
Yes, probably there is better to have a new method, right?

Reply:
:+1:

Reply:
Done!